### PR TITLE
feat: Added invalid percent publisher.

### PR DESCRIPTION
**Pull request by grieneis**

### CODE REVIEW

File: amcl/src/amcl/sensors/amcl_laser.cpp [hunk 1]
```diff
@@ -213,10 +215,11 @@ double AMCLLaser::BeamModel(AMCLLaserData *data, pf_sample_set_t* set)
     total_weight += sample->weight;
   }
 
+  *percent_invalid_poses = (float)invalid_count / set->sample_count;
```
Comment (dantwinkler) [inline amcl/src/amcl/sensors/amcl_laser.cpp]: Not sure it's possible, but we should check for set->sample_count != 0 case.

File: amcl/src/amcl/sensors/amcl_laser.cpp [hunk 2]
```diff
@@ -309,10 +315,13 @@ double AMCLLaser::LikelihoodFieldModel(AMCLLaserData *data, pf_sample_set_t* set
     total_weight += sample->weight;
   }
 
+  ROS_WARN("Invalid count %d", invalid_count);
```
Comment (dantwinkler) [inline amcl/src/amcl/sensors/amcl_laser.cpp]: I think this should be debug or trace.  It happens a lot during normal operation.

File: amcl/src/amcl/sensors/amcl_laser.cpp [hunk 3]
```diff
@@ -525,7 +535,7 @@ double AMCLLaser::LikelihoodFieldModelProb(AMCLLaserData *data, pf_sample_set_t*
 
   delete [] obs_count; 
   delete [] obs_mask;
-
+  *percent_invalid_poses = (float)invalid_count / set->sample_count;
```
Comment (dantwinkler) [inline amcl/src/amcl/sensors/amcl_laser.cpp]: Divide by 0 check.

File: amcl/src/amcl/sensors/amcl_laser.cpp [hunk 4]
```diff
@@ -309,10 +315,13 @@ double AMCLLaser::LikelihoodFieldModel(AMCLLaserData *data, pf_sample_set_t* set
     total_weight += sample->weight;
   }
 
+  ROS_WARN("Invalid count %d", invalid_count);
+  *percent_invalid_poses = (float)invalid_count / set->sample_count;
```
Comment (dantwinkler) [inline amcl/src/amcl/sensors/amcl_laser.cpp]: Divide by 0 check.

**Review by dantwinkler — CHANGES_REQUESTED** [review]
Just a few divide by 0 checks.  Don't think it's possible for it to be 0 but just in case.

**Comment by grieneis** [comment]
Updated based on code review.

**Review by dantwinkler — APPROVED** [review]

**Pull request merged**